### PR TITLE
memory/ndctl_selftest.py : Enabled the support for meson build

### PR DESCRIPTION
**Pull request by rohan-r-deshpande**
This PR addresses the below changes - 

  1] Enabled meson build for ndctl
  2] Added ndctl version in yaml file
  3] Header file checks for iniparser.h
  4] Incorporated code changes of PR#2330 : https://github.com/avocado-framework-tests/avocado-misc-tests/pull/2330
  
  5] ndctl.py : removed a conditional check that was meant for SLES15

### CODE REVIEW

File: memory/ndctl.py [hunk 1]
```diff
@@ -200,12 +200,6 @@ def setUp(self):
         self.reflink = '-m reflink=0'
         self.smm = SoftwareManager()
         if self.package == 'upstream':
-            # Skipping this for now, due to non-availibility of
-            # dependent packages.
-            if self.dist.name == 'SuSE' and int(self.dist.version) <= 15:
```
Comment (PraveenPenguin) [inline memory/ndctl.py:205]: @rohan-r-deshpande have we tried this on SLES12 or older SLES15 releases?

File: memory/ndctl_selftest.py [hunk 2]
```diff
@@ -45,6 +45,39 @@ def get_bus_ids(self):
                     bus_ids.append(value)
         return bus_ids
 
+    def autotools_build_system(self):
```
Comment (PraveenPenguin) [inline memory/ndctl_selftest.py:53]: I suggest  patch[04/03] and patch[04/05] can merge with this patch itself as having broader commit as meson build support

**Review by PraveenPenguin — CHANGES_REQUESTED** [review]

File: memory/ndctl.py (hunk 1 above)
Comment (rohan-r-deshpande) [reply memory/ndctl.py:205]: @PraveenPenguin No we haven't

Comment (rohan-r-deshpande) [reply memory/ndctl.py:205]: Done

File: memory/ndctl_selftest.py (hunk 2 above)
Comment (rohan-r-deshpande) [reply memory/ndctl_selftest.py:53]: Done

**Comment by rohan-r-deshpande** [comment]
Incorporated the review comments and here are the updated changes --
- Out of the total 5 patches committed earlier, merged patch [03/05] & patch [04/05] into patch [01/05], thereby reducing the total number of final patches to 3.

**Review by PraveenPenguin — APPROVED** [review]
LGTM

**Pull request merged**